torchtext/data/functional.py

```diff
@@ -114,3 +115,39 @@ def _internal_func(txt_iter):
         for line in txt_iter:
             yield sp_model.EncodeAsPieces(line)
     return _internal_func
+
+
+def custom_replace(replace_pattern):
+    r"""A transform to convert text string.
+
+    Examples:
+        >>> from torchtext.data.functional import custom_replace
+        >>> custom_replace_transform = custom_replace([(r'S', 's'), (r'\s+', ' ')])
+        >>> list_a = ["Sentencepiece encode  aS  pieces", "exampleS to   try!"]
+        >>> list(custom_replace_transform(list_a))
+            ['sentencepiece encode as pieces', 'examples to try!']
+    """
+
+    _patterns = list((re.compile(p), r)
+                     for (p, r) in replace_pattern)
+
+    def _internal_func(txt_iter):
```

Comment:
What's the goal of wrapping the yield inside a function? Could you remove the `def` and the `return _internal_func`?

Reply:
The `custom_replace` func returns `_internal_func`, which accepts `txt_iter` for replacement.  I could make those two as a single func but then, I have to re.compile the replace pattern every time.